### ISSUE TITLE
When navigating responses with alt+[ and alt+], the next response is not read

### ISSUE THREAD

**Issue by roblourens** [microsoft/vscode#188831]
Testing #188537

- Press alt+F2, VoiceOver starts reading the response
- Press alt+], the view shows the next response but VoiceOver is still reading that response and it never reads the next response

Maybe related, but sometimes, it also doesn't read the response when I press alt+F2. It will say "you are now on a text area" and just sit there. Then, it only wants to read the text a single character at a time when I move the cursor.

**Comment by meganrogge** [comment]
Are you in screen reader optimized mode?

**Comment by meganrogge** [comment]
Actually, I believe this might be working as expected - I have to move the cursor for the screen reader to move on to the next content. 

@jooyoungseo and @rperez030 could you pls let us know if this is a bug or working as expected?

**Comment by rperez030** [comment]
Unfortunately, this is a voiceover issue, but I believe last time we faced something like this it was resolved my slightly increasing the wait time for certain focus event. I don't remember the full context and cannot find the issue right now. @meganrogge  does it sound familiar? It was related to the opening and closing of the contextual help I believe.

**Comment by meganrogge** [comment]
Great recall @rperez030 the issue you are talking about is https://github.com/microsoft/vscode/issues/186514

**Comment by rperez030** [comment]
Correct. I believe it could be the same underlying cause. By the way, having to move the cursor to read content is totally fine as you described. the real issue is that Voiceover is not announcing the change of context.

**Comment by meganrogge** [comment]
Makes sense - because without that announcement, you wouldn't know to move your cursor

**Comment by rperez030** [comment]
I wonder if we have a way to calculate the total number of responses. if we do, we could issue an alert announcing position information when we move back and forth. This would resolve the lack of context for macOs and improve the experience in the other platforms. Thoughts?

**Comment by meganrogge** [comment]
Yes @rperez030 that makes sense to me - I linked an issue where this is a request for the notification a11y view and we could also support this for the responses

**Comment by meganrogge** [comment]
It might have to be an August thing - I'll investigate.

**Comment by meganrogge** [comment]
this is also an issue for notifications

**Comment by meganrogge** [comment]
I think we have to go with what you suggested @rperez030 i tried a bunch of hacks and setTimeouts and it didn't work

**Comment by rperez030** [comment]
> I think we have to go with what you suggested @rperez030 i tried a bunch of hacks and setTimeouts and it didn't work

I think it could make the feature more helpful for all platforms. Moving between suggestions is right now a bit confusing because I can't tell for sure if the one on focus is the last one, or if it truly moved to a different one after pressing the command. In the copilot chat panel, for example, I find much more practical and easier to simply move to the list of responses and go on from there. In addition, hacking assistive technology rarely justifies the effort, and often leads to undesirable outcomes that are not easy to anticipate.

**Comment by meganrogge** [comment]
it seems focus still gets trapped initially - even alert isn't taking precedent. i'm hoping it's just what I'm experiencing. about to make a PR with a change that does this for notifications and chat responses

**Comment by meganrogge** [comment]
yes, the linked PR please @rperez030 sorry I have been doing a lot atm 😓

**Comment by meganrogge** [comment]
Verified by @rperez030 with the follow up item of to be addressed in August https://github.com/microsoft/vscode/issues/189105
